lib/src/layout/sidebar_item.dart

```diff
@@ -0,0 +1,502 @@
+    this.disclosureColor,
+    this.shape,
+  })  : assert(currentIndex >= 0),
+        super(key: key);
+
+  /// The [SidebarItem]s used by the sidebar. If no items are provided,
+  /// the sidebar is not rendered.
+  final List<SidebarItem> items;
+
+  /// The current selected index. It must be in the range of 0 to
+  /// [items.length]
+  final int currentIndex;
+
+  /// Called when the current selected index should be changed.
+  final ValueChanged<int> onChanged;
+
+  /// The scroll controller used by this sidebar. If null, a local scroll
+  /// controller is created.
+  final ScrollController? scrollController;
+
+  /// The color to paint the item iwhen it's selected.
+  ///
+  /// If null, [MacosThemeData.primaryColor] is used.
+  final Color? selectedColor;
+
+  /// The color to paint the item when it's unselected.
+  ///
+  /// Defaults to transparent.
+  final Color? unselectedColor;
+
+  /// The color to paint the disclosure item.
+  final Color? disclosureColor;
+
+  /// The [shape] property specifies the outline (border) of the
+  /// decoration. The shape must not be null. It's used alonside
+  /// [selectedColor].
+  final ShapeBorder? shape;
+
+  List<SidebarItem> get _allItems {
+    List<SidebarItem> result = [];
+    items.forEach((element) {
+      if (element.disclosureItems != null) {
+        result.addAll(element.disclosureItems!);
+      } else {
+        result.add(element);
+      }
+    });
+    return result;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (items.isEmpty) return SizedBox.shrink();
+    assert(debugCheckHasMacosTheme(context));
+    assert(currentIndex < _allItems.length);
+    final theme = MacosTheme.of(context);
+    return IconTheme.merge(
+      data: const IconThemeData(size: 20),
+      child: _SidebarItemsConfiguration(
+        selectedColor: selectedColor ?? theme.primaryColor,
+        unselectedColor: unselectedColor ?? _transparent,
+        disclosureColor: disclosureColor ?? _defaultDisclosureColor,
+        shape: shape ?? _defaultShape,
+        child: ListView(
+          controller: scrollController,
+          physics: const ClampingScrollPhysics(),
+          padding: EdgeInsets.all(10.0 - theme.visualDensity.horizontal),
+          children: List.generate(items.length, (index) {
+            final item = items[index];
+            if (item.disclosureItems != null) {
+              return _DisclosureSidebarItem(
+                item: item,
+                selectedItem: _allItems[currentIndex],
+                onChanged: (item) {
+                  onChanged(_allItems.indexOf(item));
+                },
+              );
+            }
+            return _SidebarItem(
+              item: item,
+              selected: _allItems[currentIndex] == item,
+              onClick: () => onChanged(_allItems.indexOf(item)),
+            );
+          }),
+        ),
+      ),
+    );
+  }
+}
+
+class _SidebarItemsConfiguration extends InheritedWidget {
```

Comment:
Why is this class private?

Reply:
`SidebarItems` should be used for configuration instead of this. If you wrap a `SidebarItems` with a `_SidebarItemsConfiguration`, it'll have no effect!

Reply:
Okay